corehq/apps/app_manager/helpers/validators.py

```diff
@@ -767,7 +767,11 @@ def validate_for_build(self, validate_module):
                         self.form.get_app().get_form(form_link.form_id)
                     except FormNotFoundException:
                         errors.append(dict(type='bad form link', **meta))
-            # TODO: add error for module not found
+                else:
+                    try:
+                        self.form.get_app().get_module_by_unique_id(form_link.module_unique_id)
+                    except ModuleNotFoundException:
+                        errors.append(dict(type='bad form link', **meta))
```

Comment:
I guess `bad form link` is never showed to users anyways?

Reply:
yeah, it's only used in the massive if-else that is `build_errors.html`